src/main/java/codesquad/springcafe/Controller/UserController.java

```diff
@@ -55,41 +59,79 @@ public String userProfile(@PathVariable("userId") String userId, Model model) {
     }
 
     @GetMapping("/{userId}/form")
-    public String userUpdate(@PathVariable("userId") String userId, Model model) {
-        //FIXME
-        // - 유저 패스워드 같은 정보를 가져오는 경우 발생
+    public String update(@PathVariable("userId") String userId, Model model) {
         try {
             User user = userService.findUserById(userId);
-            model.addAttribute("user", user);
+            model.addAttribute("userId", user.getUserId());
+            model.addAttribute("email", user.getEmail());
         } catch (NoSuchElementException e) {
             //존재하지 않는 유저 가져오면 리스트로 리다이렉트
             logger.error(e.getMessage());
             return "redirect:/users";
         }
-        return "user/updateForm";
+        return "user/update";
+    }
+
+    //   todo : 유저 업데이트
+    //    로그인한 사용자는 자신의 정보를 수정할 수 있어야 한다.
+    //    이름, 이메일만 수정할 수 있으며, 사용자 아이디는 수정할 수 없다.
+    //    비밀번호가 일치하는 경우에만 수정 가능하다.
+    @PutMapping("/update")
+    public String update(@ModelAttribute User updateUser, HttpSession httpSession) {
+        String userId = (String)httpSession.getAttribute("loginUserId");
+
+        //유저 정보가 없거나, 유저아이디와 변경할 아이디가 같지 않은경우 에러페이지 출력
+        if (userId.isEmpty() || !updateUser.getUserId().equals(userId)) {
+            return "redirect:/index";
+        }
+
+        try{
+            userService.update(updateUser);
```

Comment:
```suggestion
            User updatedUser = userService.update(user);
```
왠지는 모르겠는데 오늘은 이쪽 코드가 1% 정도 더 선호되는 것 같은 느낌이에요. 그냥 기분탓일수도 있으니 참고만 하세요.